src/models/util/warning.ts

```diff
@@ -0,0 +1,34 @@
+import Serializable from "./Serializable";
+
+export enum WarningType {
+  WARNING = "warning",
+  ERROR = "error",
```

Comment:
Is there a better name for the class than `Warning` that is more encompassing of both the `WARNING` and `ERROR` subtypes? If not, `Warning` is OK

Reply:
That's a good question. Maybe "Alert" would be a better choice of words?

Reply:
I think Alert works; I think Richard had suggested 'Issue' which is also good